test/unit/stateView/document/query/validateIndexedFields.spec.js

```diff
@@ -90,4 +98,20 @@ describe('validateIndexedFields', () => {
     expect(result).to.have.lengthOf(1);
     expect(result[0]).to.equal('arrayWithObjects.anotherFlag');
   });
+
+  it('should pass query by compound index', () => {
+    const condition = [['firstName', '==', 'name'], ['$userId', '==', 123]];
+    const result = validateIndexedFields(indexedFields, condition);
+
+    expect(result).to.be.an('array');
+    expect(result).to.be.empty();
+  });
+
+  it('should fail with query by third field of compound index', () => {
```

Comment:
I would add scenarios with elementMatch

Reply:
You mean more scenarios with elementMatch?

Reply:
Yep